src/store/ChartStore.js

```diff
@@ -231,25 +231,67 @@ class ChartStore {
             }
 
             const onLayoutDataReady = () => {
-                this.restoreLayout(stxx, layoutData);
-
-                this.setActiveSymbols(active_symbols);
-
-                if (initialSymbol && !(layoutData && layoutData.symbols)) {
-                    this.changeSymbol(initialSymbol);
-                } else if (stxx.chart.symbol) {
-                    this.currentActiveSymbol = stxx.chart.symbolObject;
-                    stxx.chart.yAxis.decimalPlaces = stxx.chart.symbolObject.decimal_places;
-                    this.categorizedSymbols = this.categorizeActiveSymbols();
-                    if (onSymbolChange) { onSymbolChange(this.currentActiveSymbol); }
-                } else {
-                    this.changeSymbol(this.defaultSymbol);
-                }
-
+                const today_utc = Date.UTC(
+                        (new Date()).getUTCFullYear(),
+                        (new Date()).getUTCMonth(),
+                        (new Date()).getUTCDate(),
+                        (new Date()).getHours(),
+                        (new Date()).getMinutes(),
+                        (new Date()).getSeconds(),
+                    ),
+                    timeToEpochGMT = (time_stirng) => {
+                        let today = new Date(),
+                            time = time_stirng.split(':');
+                        return Date.UTC(today.getUTCFullYear(), today.getUTCMonth(), today.getUTCDate(), time[0], time[1], time[2]);
+                    },
+                    updateActiveSymbols = () => {
+                        api.getTradingTimes().then((data) => {
```

Comment:
Please find a way to save the result of `api.getTradingTimes()` instead of calling it every minute. This data only changes when a new day comes.

Reply:
well As @aminroosta  discuss here , I think its all ok, [slack discuss](https://binary-group.slack.com/archives/G0D2CMTTL/p1527670159000385)

Reply:
What we were discussing is _refreshing_ the list every 10 minutes. The way you refresh the list is using the result from `getTradingTimes`. My suggestion here is that because the trading times only change every start of the new day, you don't need to make an API call to the server each time you want to refresh the list every 10 minutes.